gamebook/hero.py

```diff
@@ -0,0 +1,46 @@
+from gamebook.creature import Creature
+import random
+
+
+class Hero(Creature):
+    def __init__(self, traits):
+        self.traits = traits
+        self.is_alive = True
+
+    def attack(self, trait_name):
+        """returns the attack power for the current fight sequence"""
+        for trait in self.traits:
+            if trait_name == trait.get_name():
+                attack_trait = trait
+
+        return attack_trait.get_value() + random.randint(1, 6)
+
+    def defend(self, trait_name, damage):
+        """apllies damage if the hero lost the current fight sequence"""
+        for trait in self.traits:
+            if trait_name == trait.get_name():
+                defence_trait = trait
+
+        defence_trait.set_value(defence_trait.get_value()-damage)
```

Comment:
This is how the books work?

Reply:
in the book i based this on the damage is always 2, but i think the  damage value should be chosen by the coder.
but basically this is the mechanism in the book